crates/validation/src/header_validator.rs

```diff
@@ -248,261 +334,359 @@ mod test {
         HistoryContentKey, OverlayContentKey,
     };
     use rstest::*;
+    use serde::Deserialize;
     use serde_json::Value;
     use ssz::{Decode, Encode};
     use tree_hash::TreeHash;
     use trin_utils::submodules::{
-        read_portal_spec_tests_file, read_portal_spec_tests_file_as_bytes,
+        read_portal_spec_tests_file, read_ssz_portal_spec_tests_file,
+        read_yaml_portal_spec_tests_file,
     };
 
     use super::*;
     use crate::constants::DEFAULT_PRE_MERGE_ACC_HASH;
 
     const SPEC_TESTS_DIR: &str = "tests/mainnet/history";
 
-    #[rstest]
-    #[case(1_000_001)]
-    #[case(1_000_002)]
-    #[case(1_000_003)]
-    #[case(1_000_004)]
-    #[case(1_000_005)]
-    #[case(1_000_006)]
-    #[case(1_000_007)]
-    #[case(1_000_008)]
-    #[case(1_000_009)]
-    #[case(1_000_010)]
-    #[tokio::test]
-    async fn generate_and_verify_header_with_proofs(#[case] block_number: u64) {
-        // Use fluffy's proofs as test data to validate that trin
-        // - generates proofs which match fluffy's
-        // - validates header_with_proof's
-
-        let file = read_portal_spec_tests_file(
-            "tests/mainnet/history/headers_with_proof/1000001-1000010.json",
-        )
-        .unwrap();
-        let json: Value = serde_json::from_str(&file).unwrap();
-        let header_with_proofs = json.as_object().unwrap();
-        let header_validator = get_mainnet_header_validator();
-        let obj = header_with_proofs.get(&block_number.to_string()).unwrap();
-        // Validate content_key decodes
-        let raw_ck = obj.get("content_key").unwrap().as_str().unwrap();
-        let ck = HistoryContentKey::try_from_hex(raw_ck).unwrap();
-        match ck {
-            HistoryContentKey::BlockHeaderByHash(_) => (),
-            _ => panic!("Invalid test, content key decoded improperly"),
+    mod pre_merge {
+        use super::*;
+
+        #[rstest]
+        #[case::block_number_1_000_001(1_000_001)]
+        #[case::block_number_1_000_002(1_000_002)]
+        #[case::block_number_1_000_003(1_000_003)]
+        #[case::block_number_1_000_004(1_000_004)]
+        #[case::block_number_1_000_005(1_000_005)]
+        #[case::block_number_1_000_006(1_000_006)]
+        #[case::block_number_1_000_007(1_000_007)]
+        #[case::block_number_1_000_008(1_000_008)]
+        #[case::block_number_1_000_009(1_000_009)]
+        #[case::block_number_1_000_010(1_000_010)]
+        #[tokio::test]
+        async fn generate_and_verify_header_with_proofs(#[case] block_number: u64) {
+            // Use fluffy's proofs as test data to validate that trin
+            // - generates proofs which match fluffy's
+            // - validates hwps
```

Comment:
```suggestion
        async fn generate_and_verify_header_with_proofs(#[case] block_number: u64) {
            // Use fluffy's proofs as test data to validate that trin
            // - generates proofs which match fluffy's
            // - validates header_with_proof's
```

It would be nice if we avoided the short hand in the comments at least

Reply:
I ended up refactoring all pre-merge tests:
- no more "hwp"
- proof generation part of the tests is moved into `accumulator.rs`
- removed no longer needed asset file and some constants